src/evergreen/api.py

```diff
@@ -321,6 +344,29 @@ def test_stats_by_project(self,
         test_stats_list = self._paginate(url, params)
         return [TestStats(test_stat, self) for test_stat in test_stats_list]
 
+    def tasks_by_project(self, project_id, statuses=None):
+        """
+        Get all the tasks for a project.
+
+        :param project_id: The project's id.
+        :param statuses: the types of statuses to get tasks for.
+        :return: The list of matching tasks.
+        """
+        url = self._create_url(
+            "/projects/{project_id}/versions/tasks".format(project_id=project_id))
+        params = {'status': statuses} if statuses else None
+        return [Task(json, self) for json in self._paginate(url, params)]
+
+    def project_history(self, project_id):
+        """
+        Get a project's history
+
+        :param project_id: The project's id.
+        :return: The project's history
+        """
+        url = self._create_v1_url("/projects/{project_id}/versions".format(project_id=project_id))
```

Comment:
When we can I would rather move old projects to the latest version of the API as opposed to adding support for the old APIs. In this case, there is a v2 version of this url. Would it be possible to use that instead?

Reply:
I think there may be some issue with switching; `v1`-->`v2` yields [this](https://evergreen.mongodb.com/rest/v2/projects/sys-perf/versions) instead of [this](https://evergreen.mongodb.com/rest/v1/projects/sys-perf/versions)

Reply:
Adding the parameter does help, but I think signal_processing is relying on task information under each version's builds that is not present on the v2 endpoint :/. 

Would loading `tasks_by_build` for each returned build yield the same information as before? If so that may be serviceable, but drags an N+1 load issue with it.